docs/src/main/asciidoc/getting-started-knative.adoc

```diff
@@ -24,121 +23,107 @@ The OpenShift section leverages OpenShift build and route features which are not
+We recommend to follow the instructions in the next sections and build the application step by step. However, you can go right to the completed example.
 
 Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
 
 The solution is located in the `getting-started-knative` directory.
 
-== Deploying the application in Knative
+== Set up Kubernetes Cluster
 
-Before we deploy the application to Knative in Minikube or Minishift we need to create the following Kubernetes objects:
+The following script will start minikube and deploy Knative.
 
-- https://github.com/knative/docs/blob/master/docs/serving/deploying-with-private-registry.md#provide-container-registry-credentials-to-knative[Container registry secrets] :-
-This is required to for the built container image to be pushed to the container registry of your choice
-- https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys[Deploy Key] :-
-This is required only if you are going to pull the sources from private repository.
-- https://github.com/knative/docs/tree/master/serving/samples/build-private-repo-go#setting-up-our-build-service-account[Build Service Account] :-
-The Kubernetes Service Account that will have access to Container Registry secret and Deploy Key secret
+[source,bash]
+----
+./bin/start-minikube.sh
+eval $(minikube docker-env) #<1>
+----
 
-[NOTE]
-====
-If you are not using private GitHub repo then you dont need the `Deploy Key` created and added to the Build Service Account
-====
+<1> Make the current Docker context to be that of minikube
 
-Run the following commands to kick start https://github.com/knative/build[Knative Build] which will build the quarkus application container image
-using Dockerfile and https://github.com/GoogleContainerTools/kaniko[Kaniko].  After a successful build you will have the
-Knative serving application deployed with the built container image.  You can watch the application build pods using the
-command `kubectl get pods -w`. You can terminate the watch using the command kbd:[CTRL + c]
+== Set up Nexus(Optional)
 
-Since there are some parameters that you need to pass to the builds, which are right now configurable via maven properties,
-you need to run the following maven command to make them passed to the Knative resource yamls.
+Nexus is used for caching maven artifacts so that Apache Maven builds are faster.
 
-.Maven Parameters
-|===
-|Name |Use |Example
+[source,bash]
+----
+kubectl apply -f k8s/nexus.yaml
+----
 
-| github.deploy.key
-| the base64 encoded private key that is configured to be used as the GitHub private repo Deploy key
-| `cat ~/.ssh/quarkus-quickstarts \| base64 -w 0`
+Wait for some time to have Nexus initialize and run. You can watch the status via `kubectl get pods -w`, use kbd:[Ctrl+c] to terminate the watch.
 
-| github.keyscan
-| the base64 encoded value of `ssh-keyscan github.com`
-| `ssh-keyscan github.com \| base64 -w 0`
+== Why Skaffold ?
```

Comment:
Why Skaffold? Shouldn't we offer an integral solution? Now that we have Jib support maybe the guide should be updated to use Jib? At least let's offer a single way of doing things to not confuse the users.